src/snapshot_derive/src/common.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{ATTRIBUTE_NAME, END_VERSION, START_VERSION};
```

Comment:
nit: the split into files seems quite hard to understand. `descriptor.rs` contains all the descriptors (`Struct`, `Enum`, `Union`), the field types are split into different files: `struct_field`, `enum_field`, `union_field`. `common.rs` contains mostly field related logic. 

One option would be to put in `common.rs` the traits `Field` (if you plan to cdefine it) and `Descriptor`, and then have files/folders per type: `Struct`, `Enum`, `Union`. Another option would be to have a folder for the fields and a folder fo the descriptors. I think I would prefer the first one.

Reply:
Thanks for the suggestions. I will see what works best.

Reply:
I've split the descriptor, please take a look.

Reply:
Looks good.